driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy

```diff
@@ -297,6 +299,42 @@ class AggregatesFunctionalSpecification extends OperationFunctionalSpecification
         results.first().get('res', List) == ['c']
     }
 
+    @IgnoreIf({ serverVersionLessThan(7, 0) })
```

Comment:
While testing $unset and $geoNear, I added `AggregatesTest`. I preferred this because I found groovy hard to follow, and because it did not show the code as the user would see/write it. Should we write all new tests in Java? (I understand these tests are already written, like the existing tests.)

Reply:
Thank you for sharing your perspective. Personally, I find Groovy tests to be descriptive and convenient in certain scenarios. Java tests can occasionally become bulky. However, I agree with your viewpoint that writing tests in Java brings us closer to the user experience. I will make the change and switch them to Java.

Reply:
Thank you for sharing your perspective. Personally, I find Groovy tests to be descriptive and convenient for data-driven tests. While Java tests can occasionally become bulky, I agree with your viewpoint that writing tests in Java brings us closer to the user experience. At the same time, I believe that we could still leverage vanilla Java within Groovy to achieve that goal and make test preparation and assertions with Groovy syntax.

Taking everything into consideration, I think it's best to stick with our usual practice of writing new tests in Java and avoid adding more Groovy tests to the specifications. So, I'll make the switch to Java for new test cases.